src/Desktop.Analyzers/UnitTests/DoNotUseInsecureDTDProcessingUseXmlReaderForXPathDocumentTests.cs

```diff
@@ -0,0 +1,366 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.UnitTests;
+using Microsoft.CodeAnalysis.CSharp;
+using Xunit;
+
+namespace Desktop.Analyzers.UnitTests
+{
+    public partial class DoNotUseInsecureDTDProcessingAnalyzerTests : DiagnosticAnalyzerTestBase
+    {
+        private DiagnosticResult GetCA3075XPathDocumentCSharpResultAt(int line, int column, string name)
+        {
+            return GetCSharpResultAt(line, column, CA3075RuleId, string.Format(CA3075LoadXmlMessage, name, ".ctor"));
+        }
+
+        private DiagnosticResult GetCA3075XPathDocumentBasicResultAt(int line, int column, string name)
+        {
+            return GetBasicResultAt(line, column, CA3075RuleId, string.Format(CA3075LoadXmlMessage, name, ".ctor"));
+        }
+
+        [Fact]
+        public void UseXPathDocumentShouldGenerateDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.Xml;
+using System.Xml.XPath;
+
+namespace FxCopUnsafeXml
+{
+    public class UseXmlReaderForXPathDocument
+    {
+        public void TestMethod(string path)
+        {
+            XPathDocument doc = new XPathDocument(path);
+        }
+    }
+}
+",
+                GetCA3075XPathDocumentCSharpResultAt(11, 33, "TestMethod")
+            );
+
+            VerifyBasic(@"
+Imports System.Xml
+Imports System.Xml.XPath
+
+Namespace FxCopUnsafeXml
+    Public Class UseXmlReaderForXPathDocument
+        Public Sub TestMethod(path As String)
+            Dim doc As New XPathDocument(path)
+        End Sub
+    End Class
+End Namespace",
+                GetCA3075XPathDocumentBasicResultAt(8, 24, "TestMethod")
+            );
+        }
+
+        [Fact]
+        public void UseXPathDocumentInGetShouldGenerateDiagnostic()
```

Comment:
Likewise, `XPathDocumentInPropertyGetterWithoutSecureReaderShouldGenerateDiagnostic`. Although if you're going to have a test like this, do you need one like `...InLambda...`?